src/content/docs/en/guides/upgrade-to/v6.mdx

```diff
@@ -75,6 +75,137 @@ The following features have now been entirely removed from the code base and can
 
 Projects now containing these removed features will be unable to build, and there will no longer be any supporting documentation prompting you to remove these features.
 
+
+### Removed: legacy content collections
+
+<SourcePR number="14407" title="fix: remove legacy content collections"/>
+
+In Astro 5.x, it was still possible to use [the Content Collections API first introduced in Astro v2.0](https://astro.build/blog/introducing-content-collections/), either through a `legacy` configuration flag or via built-in backwards compatibility. These methods allowed you to upgrade to Astro v5 even if you were not yet ready or able to update your content collections to the new Content Layer API.
+
+**Astro v6.0 removes this previously deprecated Content Collections API support entirely, including the `legacy.collections` flag.** All content collections must now use the Content Layer API introduced in Astro v5.0 that powers all content collections. No backwards compatibility support is available.
+
+#### What should I do?
+
+If you had previously enabled the legacy flag, you must remove it.
+
+```ts title="astro.config.mjs" del={5}
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  legacy: {
+    collections: true,
+  }
+})
+```
+
+Additionally, if you did not upgrade your collections for Astro v5.0, ensure that your content collections are **fully updated** for the updated API.
+
+Astro v5.x included some automatic backwards compatibility to allow content collections to continue to work even if they had not been updated to use the new API. Therefore, your v5 collections may contain one or more legacy features that need updating to the newer API for v6, even if your project was previously error-free.
+
+If you have content collections errors after upgrading to v6, use the following list to help you identify and upgrade any legacy features that may exist in your code.
```

Comment:
```suggestion
If you have [content collections errors](/en/reference/error-reference/#content-collection-errors) after upgrading to v6, use the following list to help you identify and upgrade any legacy features that may exist in your code.
```
The link exists, but I haven't bothered to put these individual errors on the index page. I don't know whether we need to explicitly include the 3 legacy errors here, or just point them to the content collections general set of errors.